### PR TITLE
refactor (parseCssColor): use parseInt, avoid unnecessary coercion

### PR DESCRIPTION
**Pull request by ljharb**
Upstream some changes from https://github.com/nodejs/node/pull/49205

### CODE REVIEW

**Comment by bartlomieju** [comment]
Thanks for the PR @ljharb; any chance you could update `cli/tests/node_compat/test/parallel/test-util-format.js` to include the tests from PR in Node repo?

**Comment by ljharb** [comment]
Sure, i'll have to clone the repo so give me a few days.

File: ext/console/01_console.js [hunk 1]
```diff
@@ -2742,34 +2742,34 @@ const HSL_PATTERN = new SafeRegExp(
 );
 
 function parseCssColor(colorString) {
-  if (MapPrototypeHas(colorKeywords, colorString)) {
-    colorString = MapPrototypeGet(colorKeywords, colorString);
+  if (colorKeywords.has(colorString)) {
```
Comment (lucacasonato) [inline ext/console/01_console.js:2745]: FYI to other reviewers: safe because `SafeMap`, but may result in `prefer-primordials` linter errors. Let's see :)

**Review by lucacasonato — APPROVED** [review]
LGTM

**Comment by lucacasonato** [comment]
@bartlomieju `NO_COLOR` seems to be enabled in the node compat tests, so the tests don't pass. PTAL

**Comment by ljharb** [comment]
@bartlomieju actually these are just refactors and have nothing to do with the meat of the node PR; what tests would you want me to include?

**Comment by ljharb** [comment]
any update? These are just simple refactors to improve the code quality and potentially performance.

**Review by bartlomieju — APPROVED** [review]
Sorry for a slow response @ljharb. I revert that test file change as running update scripts for Node compat tests causes them to be removed anyway. Thanks, LGTM

**Comment by ljharb** [comment]
Thanks!

**Pull request merged**